src/audio/coreaudio/SDL_coreaudio.m

```diff
@@ -1244,6 +1244,8 @@ output device (in which case we'll try again). */
 
     if (name != NULL) {
         *name = devname;
+    } else {
```

Comment:
Why are we checking name here? Can't we bypass all the work above if name is NULL? Or are we using devname elsewhere?

Reply:
I believe that was a remnant of the old code I had (which used the dev name elsewhere). I'll check it out and fix it.

Reply:
I don't _think_ it's used elsewhere, so hypothetically yes we could skip the above work if we wanted to. This at least fixes the leak that was here before, so it's still redundant but won't cause stability issues.